tests/proxy_overload_recovery.erl

```diff
@@ -0,0 +1,442 @@
+%%
+%% QuickCheck test to verify that the vnode proxy overloads when the
+%% mailbox estimate is in overload and that it recovers from overload
+%% state once the vnode has processed the message queue.  In the real
+%% system, messages are sent to the vnode outside of the proxy so this
+%% is also modelled.
+%%
+%% Properties -
+%%   1) If the vnode message queue has cleared, the vnode proxy should not
+%%   be in overload.
+%%   2) If the proxy has been sent more messages than the overload threshold
+%%      while the vnode is suspended, it should be in overload.
+%%   3) If the proxy has been sent less than or equal to the overload threshold
+%%      it should not be in overload.
+%%
+%% Whenever the vnode is resumed, the test checks the message queues are
+%% cleared and the model is reset.  The main goal is to advance the proxy
+%% into interestign new states.
+%%
+%% This test can be run outside of riak_test while working on it. 
+%% Symlink the source into a release build and run
+%%   c(proxy_overload_recovery). 
+%%   proxy_overload_recovery:run(300). % Run for 5 mins
+%%
+%% On failure you can re-run counter examples *and* print out the internal
+%% state with the run.
+%%   proxy_overload_recovery:check(). 
+%%
+%% TODO/Questions:
+%% 1) Is there a better way to do the initialization step?
+%% 2) Remove the riak_kv dependency so it's a pure riak_core test.
+
+-module(proxy_overload_recovery).
+-include_lib("eqc/include/eqc.hrl").
+-include_lib("eqc/include/eqc_statem.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-export([confirm/0]).
+-compile(export_all).
+
+confirm() ->
+    %% disable AAE to prevent vnode start delay opening AAE leveldb
+    Opts = [{riak_kv, [{anti_entropy, {off, []}}]}],
+    Nodes = [Node1] = rt:deploy_nodes(1, Opts),
```

Comment:
Most of the new tests uses the `build_clusters` API, and no longer call `deploy_nodes` directly.